proxy/agent.go

```diff
@@ -175,8 +175,14 @@ func (a *agent) Run(stop <-chan struct{}) {
 			}
 
 		case <-time.After(delay):
-			glog.V(2).Infof("Reconciling after delay %v", delay)
-			a.reconcile()
+			if reflect.DeepEqual(a.desired, a.current) {
+				glog.V(2).Info("Reconciliation not needed")
+				a.retryBudget = a.maxRetries
```

Comment:
Managers needs to exit if maxRetries exhausted, so the pod can be re-started. Having envoy not running while manager keeps running is a big issue. This can be addressed in another PR.